internal/cli/atlas/alerts/settings/validator.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package settings
+
+import (
+	"errors"
+	"fmt"
+)
+
+func validateConfigOpts(opts *ConfigOpts) error {
+	if opts.event == "" {
+		return fmt.Errorf("--event flag is required")
+	}
+	if opts.notificationIntervalMin == 0 {
+		return fmt.Errorf("--notificationIntervalMin is required")
+	}
+	if opts.notificationType == "" {
+		return fmt.Errorf("--notificationType is required")
+	}
+
+	err := validateAlertSettingsTypes(opts)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+//gocyclo:ignore
+func validateAlertSettingsTypes(opts *ConfigOpts) error {
+	switch opts.notificationType {
+	case datadog:
+		if opts.apiKey == "" || opts.notificationRegion == "" {
+			return errors.New("--apiKey and --notificationRegion are required when --notificationType is DATADOG")
+		}
```

Comment:
gocyclo goes up by having if inside an case, which is usually a smell👃 
you should move the `if`s to functions
```go
case datadog: validateVatadog()
```